lib/location/method_cellular.c

```diff
@@ -0,0 +1,195 @@
+static struct lte_lc_cells_info cell_data = {
+	.neighbor_cells = neighbor_cells,
+};
+static bool running;
+
+void method_cellular_lte_ind_handler(const struct lte_lc_evt *const evt)
+{
+	switch (evt->type) {
+	case LTE_LC_EVT_NEIGHBOR_CELL_MEAS: {
+		struct lte_lc_cells_info cells = evt->cells_info;
+		struct lte_lc_cell cur_cell = cells.current_cell;
+
+		LOG_DBG("Cell measurements results received");
+		memset(&cell_data, 0, sizeof(struct lte_lc_cells_info));
+
+		if (cur_cell.id) {
+			/* Copy current and neighbor cell information */
+			memcpy(&cell_data, &evt->cells_info, sizeof(struct lte_lc_cells_info));
+			memcpy(neighbor_cells, evt->cells_info.neighbor_cells,
+				sizeof(struct lte_lc_ncell) * cell_data.ncells_count);
+			cell_data.neighbor_cells = neighbor_cells;
+		} else {
+			LOG_INF("No current cell information from modem.");
+		}
+		if (!cells.ncells_count) {
+			LOG_INF("No neighbor cell information from modem.");
+		}
+		k_sem_give(&cellmeas_data_ready);
+	} break;
+	default:
+		break;
+	}
+}
+
+static int method_cellular_ncellmeas_start(void)
+{
+	int err;
+
+	LOG_DBG("Triggering start of cell measurements");
+	err = lte_lc_neighbor_cell_measurement();
+	if (err) {
+		LOG_ERR("Failed to initiate neighbor cell measurements: %d", err);
+		loc_core_event_cb_error();
+		running = false;
+		return err;
+	}
+	return 0;
+}
+
+static void method_cellular_positioning_work_fn(struct k_work *work)
+{
+	struct multicell_location location;
+	struct loc_location location_result = { 0 };
+	int ret;
+	struct k_work_args *work_data = CONTAINER_OF(work, struct k_work_args, work_item);
+	const struct loc_cellular_config cellular_config = work_data->cellular_config;
+
+	loc_core_timer_start(cellular_config.timeout);
+
+	ret = method_cellular_ncellmeas_start();
+	if (ret) {
+		LOG_WRN("Cannot start cell measurements - cannot fetch location");
+		loc_core_event_cb_error();
+		running = false;
+		return;
+	}
+
+	ret = k_sem_take(&cellmeas_data_ready, K_SECONDS(60)); /* TODO: timeout to params? */
+	if (ret) {
+		LOG_WRN("Timeout for waiting cell measurements - cannot fetch location");
+		loc_core_event_cb_error();
+		running = false;
+	} else if (running) {
+		if (cell_data.current_cell.id == 0) {
+			LOG_WRN("No cells were found - cannot fetch location");
+			loc_core_event_cb_error();
+			running = false;
+			return;
+		}
+
+		/* enum multicell_service can be used directly
+		 * because BUILD_ASSERT verifies enums are equal
+		 */
+		ret = multicell_location_get(cellular_config.service, &cell_data, &location);
+		if (ret) {
+			LOG_ERR("Failed to acquire location from multicell_location lib, error: %d",
+				ret);
+			loc_core_event_cb_error();
+			running = false;
+		} else {
+			location_result.latitude = location.latitude;
+			location_result.longitude = location.longitude;
+			location_result.accuracy = location.accuracy;
+			(void)lte_lc_neighbor_cell_measurement_cancel();
```

Comment:
This doesn't need to be called if we've received the measurements. Each AT%NCELLMEAS command only returns a single report and then stops.